### PR TITLE
SG-17191 Added the ability to flag certain includes as optional

### PR DESCRIPTION
**Pull request by eshokrgozar**
Allows the user to flag an include as not required (optional). If an optional include resolves to a missing file, a warning is logged instead of raising an Exception. 

Here's the new supported syntax (it still supports the old way too):
```
includes:
- {path: ./path/to/the_file.yml, required: false}
```

### CODE REVIEW

File: tests/platform_tests/test_environment.py [hunk 1]
```diff
@@ -105,6 +106,37 @@ def test_app_meta(self):
             self.raw_app_metadata["configuration"],
         )
 
+    def test_missing_include(self):
```
Comment (jfboismenu) [inline tests/platform_tests/test_environment.py:109]: Given what I've mentioned above for the refactoring of the includes computation code to `includes.get_includes` method, you can probably move these tests to `test_includes.py`

File: python/tank/util/includes.py [hunk 2]
```diff
@@ -45,6 +50,59 @@ def _is_current_platform_abspath(path):
         return posixpath.isabs(path)
 
 
+def _get_includes(file_name, data):
```
Comment (jfboismenu) [inline python/tank/util/includes.py:53]: Please rename to `get_includes`

File: tests/core_tests/test_includes.py [hunk 3]
```diff
@@ -11,191 +11,217 @@
...
-    def _resolve_includes(self, includes):
+        # We do permutations here because Python 2 and Python 3 handle
+        # set ordering differently.
+        for includes in itertools.permutations(["a.yml", "b.yml", "c.yml"]):
+            self.assertEqual(
+                self._resolve_includes(includes),
+                [os.path.join(os.getcwd(), include) for include in includes],
+            )
+
+    def test_missing_include(self):
```
Comment (jfboismenu) [inline tests/core_tests/test_includes.py:169]: Please add missing docstrings.

**Comment by pmacdonald-falcons** [comment]
Did this end up getting released?

**Pull request closed without merging**